src/main/java/practice/StreamPractice.java

```diff
@@ -50,7 +70,13 @@ public List<Person> selectMenByAge(List<Person> peopleList, int fromAge, int toA
      */
     public List<Person> getWorkablePeople(int fromAge, int femaleToAge,
                                           int maleToAge, List<Person> peopleList) {
-        return Collections.emptyList();
+        Predicate<Person> manPredicate = p -> p.getSex() == Person.Sex.MAN
+                && p.getAge() >= fromAge && p.getAge() <= maleToAge;
+        Predicate<Person> womanPredicate = p -> p.getSex() == Person.Sex.WOMAN
+                && p.getAge() >= fromAge && p.getAge() <= femaleToAge;
+        return peopleList.stream()
+                .filter(manPredicate.or(womanPredicate))
```

Comment:
you can simplify the predicates
```suggestion
        Predicate<Person> manAgePredicate = p -> p.getSex() == Person.Sex.MAN && p.getAge() <= maleToAge;
        Predicate<Person> womanAgePredicate = p -> p.getSex() == Person.Sex.WOMAN && p.getAge() <= femaleToAge;
        return peopleList.stream()
                .filter(p -> p.getAge() >= fromAge)
                .filter(manAgePredicate.or(womanAgePredicate))
```